File: backend/app/schemas/application_prep.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class ApplicationPrepRequest(BaseModel):
    model_config = ConfigDict(extra="forbid")

    update_decision: bool = True
    include_cold_dm_angle: bool = True
    include_resume_focus: bool = True
    include_checklist: bool = True


class ApplicationPrepListItem(BaseModel):
    label: str
    value: str
    reason: str


class ApplicationPrepResponse(BaseModel):
    job_id: str
    decision_id: str | None = None
    company_name: str | None = None
    title: str
    match_tier: str | None = None
    total_score: float | None = None
    remote_eligibility: str | None = None
    fit_summary: str
    resume_focus_points: list[ApplicationPrepListItem]
    project_talking_points: list[ApplicationPrepListItem]
    concerns: list[ApplicationPrepListItem]
    missing_information: list[str]
    suggested_next_action: str
    cold_dm_angle: str | None = None
    application_checklist: list[ApplicationPrepListItem]
    generated_at: datetime

File: frontend/types/application-prep.ts
export type ApplicationPrepRequest = {
  update_decision?: boolean;
  include_cold_dm_angle?: boolean;
  include_resume_focus?: boolean;
  include_checklist?: boolean;
};

export type ApplicationPrepListItem = {
  label?: string;
  value?: string;
  reason?: string;
};

export type ApplicationPrepResponse = {
  job_id: string;
  decision_id?: string | null;
  company_name?: string | null;
  title?: string | null;
  match_tier?: string | null;
  total_score?: number | null;
  remote_eligibility?: string | null;
  fit_summary?: string | null;
  resume_focus_points?: ApplicationPrepListItem[];
  project_talking_points?: ApplicationPrepListItem[];
  concerns?: ApplicationPrepListItem[];
  missing_information?: string[];
  suggested_next_action?: string | null;
  cold_dm_angle?: string | null;
  application_checklist?: ApplicationPrepListItem[];
  generated_at?: string;
};
